File: lda/src/main/scala/sa/Driver.scala
package scala_version

import org.apache.spark.SparkConf
import org.apache.spark.SparkContext
import org.apache.spark.SparkContext._

object Driver {
  def main(args: Array[String]): Unit = {
    val inputFile = "C:\\Users\\Junjie\\Desktop\\spark2\\src\\main\\scala\\scala_version\\Driver.scala"
    val outputFile = "C:\\Users\\Junjie\\Desktop\\data\\tmp2"

    val conf = new SparkConf().setMaster("local").setAppName("wordCount")
    val sc = new SparkContext(conf)

    val input = sc.textFile(inputFile)
    val words = input.flatMap(line => line.split(" "))
    val counts = words.map(word => (word, 1)).reduceByKey({ case (x, y) => x + y })
    counts.saveAsTextFile(outputFile)
  }
}

File: lda/src/main/java/ja/Driver.java
package ja;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaPairRDD;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.FlatMapFunction;
import org.apache.spark.api.java.function.Function2;
import org.apache.spark.api.java.function.PairFunction;
import scala.Tuple2;

import java.util.Arrays;

public class Driver {
    public static void main(String[] args) {
        SparkConf conf = new SparkConf().setMaster("local").setAppName("wordCount");
        JavaSparkContext sc = new JavaSparkContext(conf);

        // 读取输入数据
        JavaRDD<String> input = sc.textFile("C:\\Users\\Junjie\\Desktop\\big_data\\src\\main\\java\\lda\\ja.Driver.java");

        // 切分为单词
        JavaRDD<String> words = input.flatMap((FlatMapFunction<String, String>) s -> Arrays.asList(s.split(" ")));

        // 转换为键值对并计数
        JavaPairRDD<String, Integer> counts = words
                .mapToPair((PairFunction<String, String, Integer>) s -> new Tuple2(s, 1))
                .reduceByKey((Function2<Integer, Integer, Integer>) (integer, integer2) -> integer + integer2);
        counts.saveAsTextFile("C:\\Users\\Junjie\\Desktop\\data\\tmp");
    }
}
